base/stats.go

```diff
@@ -1402,6 +1431,19 @@ func (d *DbStats) unregisterDatabaseStats() {
 	prometheus.Unregister(d.DatabaseStats.SyncFunctionTime)
 }
 
+func (d *DbStats) CollectionStat(scopeName, collectionName string) *CollectionStats {
+	scopeAndCollectionName := scopeName + "." + collectionName
+	if _, ok := d.CollectionStats[scopeAndCollectionName]; !ok {
+		// DbStats was not initialised with this collection upfront in NewDBStats for some reason (_default._default for example, since we pass named collections only ...)
```

Comment:
This looks like a potential data race, and locking would be expensive.  I feel like we should rely on NewDBStats to do the right thing and just log an error if the requested scope and collection is not found.